internal/artifacts/module_arch_test.go

```diff
@@ -848,32 +843,41 @@ module-types:
 		})
 
 		It("sanity", func() {
-			modules, err := getRequiredModules(mtaObj, "m1")
+			collection := make(map[string]bool)
+			err := collectSelectedModulesAndDependencies(mtaObj, collection, "m1")
 			Ω(err).Should(Succeed())
-			Ω(modules).Should(Equal([]string{"m4", "m3", "m2"}))
+			validateMapKeys(collection, []string{"m1", "m4", "m3", "m2"})
 		})
 		DescribeTable("failures", func(targetModule string) {
-			_, err := getRequiredModules(mtaObj, targetModule)
+			collection := make(map[string]bool)
+			err := collectSelectedModulesAndDependencies(mtaObj, collection, targetModule)
 			Ω(err).Should(HaveOccurred())
 		},
 			Entry("fails on none existing target module", "m5"),
 			Entry("fails on none existing module required by the target module", "n1"),
 			Entry("fails on none existing module required by the module that is required by the target module", "n2"))
 	})
 
-	Describe("sortSelectedModules", func() {
+	Describe("sortModules", func() {
 		It("sanity", func() {
 			mtaObj := getMtaObj("mtahtml5", "mtaWithBuildRequirements.yaml")
 			allModulesSorted, err := buildops.GetModulesNames(mtaObj)
 			Ω(err).Should(Succeed())
-			selectedModules := []string{"n1", "m1"}
-			selectedModulesMap := convertListToMap(selectedModules)
-			selectedModulesSorted := sortSelectedModules(allModulesSorted, selectedModulesMap)
+			selectedModulesMap := map[string]bool{"n1": true, "m1": true}
+			selectedModulesSorted := sortModules(allModulesSorted, selectedModulesMap)
 			Ω(selectedModulesSorted).Should(Equal([]string{"m1", "n1"}))
 		})
 	})
 })
 
+func validateMapKeys(actualMap map[string]bool, expectedKeys []string) {
```

Comment:
Use `ConsistOf` matcher on the map keys instead of a custom validation function

Reply:
i don't know how to get map keys not using loop. didn't find (

Reply:
You can get the map keys with a loop and use the matcher on the result

Reply:
It can still be in the same function

Reply:
of course I can, but if I'm looping I can check already. don't see the reason why not.

Reply:
The error message will be better

Reply:
aa... good, done